device/iot-device-samples/pnp-device-sample/TemperatureController/src/main/java/samples/com/microsoft/azure/sdk/iot/TemperatureController.java

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.deps.twin.TwinCollection;
+import com.microsoft.azure.sdk.iot.device.*;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.*;
+import com.microsoft.azure.sdk.iot.pnphelpers.PnpHelper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Map.*;
+
+@Slf4j
+public class TemperatureController {
+
+    public enum StatusCode {
+        COMPLETED (200),
+        IN_PROGRESS (202),
+        NOT_FOUND (404);
+
+        private final int value;
+        StatusCode(int value) {
+            this.value = value;
+        }
+    }
+
+    // DTDL interface used: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json
+    private static final String deviceConnectionString = System.getenv("IOTHUB_DEVICE_CONNECTION_STRING");
+    private static final String MODEL_ID = "dtmi:com:example:TemperatureController;1";
+    private static final String THERMOSTAT_1 = "thermostat1";
```

Comment:
I would rather call this Component1

Reply:
This is an identifier that you would tend to use in a sample, and model the properties and commands accordingly. So, in this case, it made sense to call this `Thermostat1`, since it signifies that the component getting modelled is a `Thermostat`, and the index is 1. 
If I call this `Component`1, then my property update handlers, command handlers etc will lose context as to what interface `Component1` actually implements.
Does that make sense?